soc/st/stm32/stm32f7/Kconfig.soc

```diff
@@ -0,0 +1,61 @@
+# ST Microelectronics STM32F7 MCU line
+
+# Copyright (c) 2018 Yurii Hamann
+# Copyright (c) 2022, Rtone.
+# Copyright (c) 2023, Rahul Arasikere.
+# Copyright (c) 2023 Evan Perry Grove
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_STM32F7X
+	bool
+	select SOC_FAMILY_STM32
+
+config SOC_SERIES
+	default "stm32f7" if SOC_SERIES_STM32F0X
```

Comment:
It links with the field in soc.yml and can be used for filtering. In the future it will also be possible to use it from sysbuild, so you could have an option which can only be used on arm cortex m3's, or only on stm32f1's, etc.